files/nagios/plugins/check_openstack_interface.py

```diff
@@ -0,0 +1,179 @@
+#!/usr/bin/env python3
+
+# Copyright (C) 2021 Canonical Ltd.
+
+# Authors:
+#   Robert Gildein <robert.gildein@canonical.com>
+
+import argparse
+from configparser import ConfigParser
+
+import openstack
+from nagios_plugin3 import try_check, UnknownError, CriticalError
+
+SEPARATOR = ","
+ACTIVE = "ACTIVE"
+DOWN = "DOWN"
+VALID_STATUS = [ACTIVE, DOWN]
+INTERFACE = {
+    "network": "network.networks",
+    "floating-ip": "network.ips",
+    "server": "compute.servers",
+    "port": "network.ports",
+    "security-group": "network.security_groups",
+    "subnet": "network.subnets",
+}
+INTERFACE_BY_EXISTENCE = ["security-group", "subnet"]
+
+
+def _rgetattr(obj, *attrs):
+    if len(attrs) == 0:
+        return obj
+
+    return _rgetattr(getattr(obj, attrs[0]), *attrs[1:])
+
+
+def _parse_arguments():
+    """Parse the check arguments and connect to OpenStack.
+
+    :returns: credentials, interface name, set IDs,
+              set IDs to skip, values to filter when using `--all`
+              and check all flag
+    :rtype: Tuple[configparser.ConfigParser, str, set, set, dict, bool]
+    """
+    credentials = ConfigParser()
+    parser = argparse.ArgumentParser("check_openstack_interface")
+    parser.add_argument("interface", type=str, help="interface name")
+    parser.add_argument("-c", "--credential", required=True,
+                        type=argparse.FileType("r"),
+                        help="path to OpenStack credential cnf file")
+    parser.add_argument("--all", action="store_true", help="check all")
+    parser.add_argument("-i", "--id", action="append", type=str, default=[],
+                        help="check specific id")
```

Comment:
The help description should mention that the flag `-i <id>` can be used more than once.